plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/WatchManager.java

```diff
@@ -271,9 +331,9 @@ public void connectionRemoved(IConnection connection) {
 			Connection conn = (Connection)connection;
 			synchronized (watches) {
 				watches.keySet().stream()
-					.filter(p->conn.ownsResource(p))
+					.filter(k->k.connection.equals(conn))
```

Comment:
connection.equals deals with URL and basic authentication info (so likely will not work with OAuth). Wondering if == should be used